master/internal/kubernetes/pod.go

```diff
@@ -237,14 +237,14 @@ func (p *pod) receivePodStatusUpdate(ctx *actor.Context, msg podStatusUpdate) er
 		if p.container.State == container.Terminated {
 			return nil
 		}
-		p.container = p.container.Transition(container.Terminated)
```

Comment:
blocking: can you explain a bit more on why is it related to the trial being stuck in Active?

Reply:
Previously if we hit an error in `getExitCodeAndMessage` our container would already be in a `Terminated` state, which meant that `finalizeTaskState()`  would think that the task state is cleaned up, and never notify the task handler & scheduler that the container terminated, leading it to be stuck in Active.

Reply:
Done